modules/nuxeo-apidoc-webengine/src/main/java/org/nuxeo/apidoc/sitemap/SitemapGenerator.java

```diff
@@ -0,0 +1,100 @@
+import static org.nuxeo.apidoc.browse.ApiBrowserConstants.LIST_PACKAGES;
+import static org.nuxeo.apidoc.browse.ApiBrowserConstants.LIST_SERVICES;
+
+import java.io.Writer;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Marshaller;
+
+import org.apache.commons.lang3.StringUtils;
+import org.nuxeo.apidoc.snapshot.DistributionSnapshot;
+import org.nuxeo.apidoc.snapshot.SnapshotManager;
+import org.nuxeo.common.utils.URIUtils;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.runtime.api.Framework;
+
+/**
+ * Generates a sitemap for available distributions.
+ *
+ * @since 20.3.0
+ */
+public class SitemapGenerator {
+
+    protected static final List<String> PATHS = List.of("", LIST_BUNDLEGROUPS, LIST_BUNDLES, LIST_COMPONENTS,
+            LIST_CONTRIBUTIONS, LIST_EXTENSIONPOINTS, LIST_OPERATIONS, LIST_PACKAGES, LIST_SERVICES);
+
+    public static void generateXML(String baseUrl, CoreSession session, Writer writer) throws JAXBException {
+        Sitemap map = getSiteMap(baseUrl, session);
+        JAXBContext context = JAXBContext.newInstance(Sitemap.class);
+        Marshaller mar = context.createMarshaller();
+        mar.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, Boolean.TRUE);
+        mar.marshal(map, writer);
+    }
+
+    protected static Sitemap getSiteMap(String baseUrl, CoreSession session) {
+        List<SitemapUrl> urls = new ArrayList<>();
+        Framework.getService(SnapshotManager.class)
+                 .getAvailableDistributions(session)
+                 .forEach(s -> urls.addAll(getUrls(baseUrl, s)));
+        return new Sitemap(urls);
+
+    }
+
+    protected static List<SitemapUrl> getUrls(String baseUrl, DistributionSnapshot snapshot) {
+        final Date date = snapshot.getReleaseDate() == null ? snapshot.getCreationDate() : snapshot.getReleaseDate();
+        String encodedDistName = snapshot.isLive() ? SnapshotManager.DISTRIBUTION_ALIAS_CURRENT
+                : URIUtils.quoteURIPathComponent(snapshot.getKey(), true);
+        return PATHS.stream()
+                    .map(p -> getUrl(baseUrl, encodedDistName, p))
+                    .map(url -> new SitemapUrl(url, date, null, null))
+                    .collect(Collectors.toList());
+    }
+
+    protected static String getUrl(String baseUrl, String encodedDistName, String... fragments) {
+        StringBuilder builder = new StringBuilder();
+        builder.append(baseUrl);
+        if (!baseUrl.endsWith("/")) {
+            builder.append("/");
+        }
+        builder.append(encodedDistName);
+        builder.append("/");
+        if (fragments != null) {
+            builder.append(StringUtils.join(
+                    Arrays.stream(fragments).map(f -> URIUtils.quoteURIPathComponent(f, true)).toArray(), "/"));
```

Comment:
We should use `String#join` instead of `StringUtils#join`.

Another solution is to do a full stream approach, for example:
```java
Arrays.stream(fragments).map(f -> URIUtils.quoteURIPathComponent(f, true)).collect(Collectors.collectingAndThen(Collectors.joining("/"), builder::append));
```
Or:
```java
builder.append(Arrays.stream(fragments).map(f -> URIUtils.quoteURIPathComponent(f, true)).collect(Collectors.joining("/")));
```